sklearn/ensemble/_forest.py

```diff
@@ -2653,3 +2654,21 @@ def transform(self, X):
         """
         check_is_fitted(self)
         return self.one_hot_encoder_.transform(self.apply(X))
+
+    def get_feature_names_out(self, input_features=None):
+        """Get output feature names for transformation.
+        Parameters
+        ----------
+        input_features : array-like of str or None, default=None
+            Not used, present here for API consistency by convention.
+        Returns
+        -------
+        feature_names_out : ndarray of str objects
+            Transformed feature names.
+        """
+        _check_feature_names_in(self, input_features)
+        class_name = self.__class__.__name__.lower()
+        return np.asarray(
+            [f"{class_name}{i}" for i in range(self.n_outputs_)],
+            dtype=object,
+        )
```

Comment:
Alternatively, you could merge the current `main` into this branch to reuse the `_ClassNamePrefixFeaturesOutMixin` mixin-class introduced in https://github.com/scikit-learn/scikit-learn/pull/21334/files#diff-ac9c452d8c660d2516089c8448596912c22417c7c690f079715593e61ef6ee0dR884

This would require you to set the private `_n_features_out` attribute at fit time.

Reply:
Hi @ogrisel , I've changed all of the classes to use `_ClassNamePrefixFeaturesOutMixin` as you suggested